docs/user_guide.rst

```diff
@@ -591,44 +592,79 @@ From within a real python, where ``SomePackage`` *is* installed globally, and is
 Ensuring Repeatability
 **********************
 
-Four things are required to fully guarantee a repeatable installation using requirements files.
+pip can achieve various levels of repeatability:
+
+Pinned Version Numbers
+----------------------
+
+Pinning the versions of your dependencies in the requirements file
+protects you from bugs or incompatibilities in newly released versions::
 
-1. The requirements file was generated by ``pip freeze`` or you're sure it only
-   contains requirements that specify a specific version.
+    SomePackage == 1.2.3
+    DependencyOfSomePackage == 4.5.6
 
-2. The installation is performed using :ref:`--no-deps <install_--no-deps>`.
-   This guarantees that only what is explicitly listed in the requirements file is
-   installed.
+Using :ref:`pip freeze` to generate the requirements file will ensure that not
+only the top-level dependencies are included but their sub-dependencies as
+well, and so on. Perform the installation using :ref:`--no-deps
+<install_--no-deps>` for an extra dose of insurance against installing
+anything not explicitly listed.
 
-3. None of the packages to be installed utilize the setup_requires keyword. See
-   :ref:`Controlling setup_requires<controlling-setup-requires>`.
+This strategy is easy to implement and works across OSes and architectures.
+However, it trusts PyPI, its CDN, and the network. It also relies on indices
```

Comment:
I'd probably drop "the network" from here because TLS means we don't need to trust the network. It's probably also reasonable to drop "CDN" too because to the average end user, the CDN is part of PyPI.